File: Leetcode 1801-1900/1828/1828.java
// Leetcode 1828: Queries on Number of Points Inside a Circle
// https://leetcode.com/problems/queries-on-number-of-points-inside-a-circle/
// Solved on 8th of November, 2025
class Solution {
    /**
     * For each query circle, counts how many points fall within or on the boundary of that circle.
     *
     * @param points A 2D array where each `points[i] = [xi, yi]` represents a point.
     * @param queries A 2D array where each `queries[j] = [xj, yj, rj]` represents a circle with center `(xj, yj)` and radius `rj`.
     * @return An array `answer` where `answer[j]` is the number of points inside the `j`-th circle.
     */
    public int[] countPoints(int[][] points, int[][] queries) {
        int qLen = queries.length;
        int pLen = points.length;
        int[] result = new int[qLen];
        for (int i = 0; i < qLen; i++) {
            long centerX = queries[i][0];
            long centerY = queries[i][1];
            long radiusSq = (long) queries[i][2] * queries[i][2];
            int count = 0;
            for (int j = 0; j < pLen; j++) {
                long dx = points[j][0] - centerX;
                long dy = points[j][1] - centerY;
                if (dx * dx + dy * dy <= radiusSq) {
                    count++;
                }
            }
            result[i] = count;
        }
        return result;
    }
}

File: Leetcode 1801-1900/1828/1828.py
# Leetcode 1828: Queries on Number of Points Inside a Circle
# https://leetcode.com/problems/queries-on-number-of-points-inside-a-circle/
# Solved on 8th of November, 2025
class Solution:
    def countPoints(self, points: list[list[int]], queries: list[list[int]]) -> list[int]:
        """
        Calculates the number of points that lie inside or on the boundary of each given circle.

        Args:
            points: A list of points, where each point is represented as [x, y].
            queries: A list of queries, where each query is represented as [centerX, centerY, radius].

        Returns:
            A list of integers, where each integer is the count of points inside the corresponding circle.
        """
        result = []

        for query in queries:
            centerX = query[0]
            centerY = query[1]
            radius = query[2]

            count = 0
            radiusSquared = radius * radius

            for point in points:
                pointX = point[0]
                pointY = point[1]

                deltaX = pointX - centerX
                deltaY = pointY - centerY

                distSquared = (deltaX * deltaX) + (deltaY * deltaY)

                if distSquared <= radiusSquared:
                    count += 1

            result.append(count)

        return result